src/Map.cpp

```diff
@@ -63,4 +78,93 @@ void Map::addTile(unsigned long id, int x, int y, GameInternal* game, const std:
 	
 	if(tile.getComponent<TileComponent>().hasCollision()) tile.addComponent<ColliderComponent>(tile.getComponent<TileComponent>().getName().data());
 	tile.addGroup((size_t)Entity::GroupLabel::MAPTILES);
+}
+
+void Map::loadMapTmx(const char* path)
+{
+    tmx::Map map;
+    if (!map.load(path)) {
+        // TODO: log to console
+    }
+
+    const std::vector<tmx::Tileset>& tileSets = map.getTilesets();
+
+    const std::vector<tmx::Layer::Ptr>& mapLayers = map.getLayers();
+    const auto mapSize = map.getTileCount();
+    const auto mapTileSize = map.getTileSize();
+
+    std::vector<std::string> texturePaths = {};
+
+    for (auto tileSet : tileSets) {
+        texturePaths.emplace_back(tileSet.getImagePath());
+    }
+
+    for (auto& layer : mapLayers) {
+
+        if (layer->getType() == tmx::Layer::Type::Tile) {
+            auto& tileLayer = layer->getLayerAs<tmx::TileLayer>();
+
+            int zIndex = 0;
+
+            const std::vector<tmx::Property>& properties = layer->getProperties();
+            auto zIndexIterator = std::find_if(properties.begin(), properties.end(), [](const tmx::Property& property) {
+                return property.getName() == "zIndex";
+            });
+
+            if (zIndexIterator != properties.end() && std::is_same<decltype(zIndexIterator->getType()), int>::value) {
+                zIndex = zIndexIterator->getIntValue();
+            }
+
+            const auto& tiles = tileLayer.getTiles();
+            
+            for (auto i = 0u; i < tileSets.size(); i++) {
+                auto tilesetTexture = VEGO_Game().textureManager->loadTexture(texturePaths.at(i).c_str());
+                int texX, texY;
+                SDL_QueryTexture(tilesetTexture, nullptr, nullptr, &texX, &texY);
+
+                const auto tileCountX = texX / mapTileSize.x;
+                const auto tileCountY = texY / mapTileSize.y;
+
+                for (auto idx = 0ul; idx < mapSize.x * mapSize.y; idx++) {
+
+                    if (!(idx < tiles.size() && tiles[idx].ID >= tileSets.at(i).getFirstGID()
+                        && tiles[idx].ID < (tileSets.at(i).getFirstGID() + tileSets.at(i).getTileCount()))) {
```

Comment:
not statement for exit/continue conditions makes it harder to read if they are not necessary